crytic_compile/compilation_unit.py

```diff
@@ -90,21 +66,62 @@ def crytic_compile(self) -> "CryticCompile":
         """
         return self._crytic_compile
 
-    ###################################################################################
-    ###################################################################################
-    # region Natspec
-    ###################################################################################
-    ###################################################################################
+    @property
+    def source_units(self) -> Dict[Filename, SourceUnit]:
+        """
+        Return the dict of the source units
+
+        Returns:
+            Dict[Filename, SourceUnit]: the source units
+        """
+        return self._source_units
+
+    def source_unit(self, filename: Filename) -> SourceUnit:
+        """
+        Return the source unit associated to the filename.
+        The source unit must have been created by create_source_units
+
+        Args:
+            filename: filename of the source unit
+
+        Returns:
+            SourceUnit: the source unit
+        """
+        return self._source_units[filename]
 
     @property
-    def natspec(self) -> Dict[str, Natspec]:
-        """Return the natspec of the contracts
+    def asts(self) -> Dict[str, Dict]:
+        """
+        Return all the asts from the compilation unit
 
         Returns:
-            Dict[str, Natspec]: Contract name -> Natspec
+            Dict[str, Dict]: absolute path -> ast
         """
-        return self._natspec
+        return {
+            source_unit.filename.absolute: source_unit.ast
+            for source_unit in self.source_units.values()
+        }
 
+    def create_source_units(self, filename: Filename) -> SourceUnit:
```

Comment:
I would rename this to make it clear it doesn't accept a list
```suggestion
    def create_source_unit(self, filename: Filename) -> SourceUnit:
```